common/djangoapps/dark_lang/api.py

```diff
@@ -0,0 +1,200 @@
+    are now deprecated, as discussed here: https://code.djangoproject.com/ticket/18419.
+    The new language codes 'zh-hans' and 'zh-hant' are now used since django 1.7.
+    Although majority of browsers still use the old language codes, some new browsers
+    such as IE11 in Windows 8.1 start to use the new ones, which makes the current
+    chinese translations of edX don't work properly under these browsers.
+    This function can keep compatibility between the old and new language codes. If one
+    day edX uses django 1.7 or higher, this function can be modified to support the old
+    language codes until there are no browsers use them.
+    '''
+    browser_langs = parse_accept_lang_header(accept)
+    django_langs = []
+    for lang, priority in browser_langs:
+        lang = CHINESE_LANGUAGE_CODE_MAP.get(lang.lower(), lang)
+        django_langs.append((lang, priority))
+
+    return django_langs
+
+# If django 1.7 or higher is used, the right-side can be updated with new-style codes.
+CHINESE_LANGUAGE_CODE_MAP = {
+    # The following are the new-style language codes for chinese language
+    'zh-hans': 'zh-CN',     # Chinese (Simplified),
+    'zh-hans-cn': 'zh-CN',  # Chinese (Simplified, China)
+    'zh-hans-sg': 'zh-CN',  # Chinese (Simplified, Singapore)
+    'zh-hant': 'zh-TW',     # Chinese (Traditional)
+    'zh-hant-hk': 'zh-HK',  # Chinese (Traditional, Hongkong)
+    'zh-hant-mo': 'zh-TW',  # Chinese (Traditional, Macau)
+    'zh-hant-tw': 'zh-TW',  # Chinese (Traditional, Taiwan)
+    # The following are the old-style language codes that django does not recognize
+    'zh-mo': 'zh-TW',       # Chinese (Traditional, Macau)
+    'zh-sg': 'zh-CN',       # Chinese (Simplified, Singapore)
+}
+LANGUAGE_INPUT_FIELD = 'preview_lang'
+RESET_SUBMIT = 'reset'
+SET_LANGUAGE_SUBMIT = 'set_language'
+
+class Darklang(object):
+    """
+    Support class and code for dark-launching languages.
+
+    This is configured by creating ``DarkLangConfig`` rows in the database,
+    using the django admin site.
+    """
+
+    @property
+    def released_langs(self):
+        """
+        Current list of released languages
+        """
+        language_options = DarkLangConfig.current().released_languages_list
+        if settings.LANGUAGE_CODE not in language_options:
+            language_options.append(settings.LANGUAGE_CODE)
+        return language_options
+
+    def process_darklang_request(self, request):
+        """
+        Prevent user from requesting un-released languages
+        """
+        if not DarkLangConfig.current().enabled:
+            return None
+
+        if RESET_SUBMIT in request.POST:
+            # Reset and clear the language preference
+            return 'Language reset to %s' % self.clear_preview_language(request)
+        if SET_LANGUAGE_SUBMIT in request.POST:
+            # Set the Preview Language
+            return self.set_preview_language(request)
+
+
+
```

Comment:
Note: set_preview_language and clear_preview_language are new methods for dealing with the incoming form data.